monkey/monkey_island/cc/setup/mongo/mongo_db_process.py

```diff
@@ -29,7 +29,10 @@ def start(self):
                 self._mongo_run_cmd, stderr=subprocess.STDOUT, stdout=log
             )
 
-        logger.info("MongoDB launched successfully!")
+        if self.is_running():
+            raise MongoDbProcessException
```

Comment:
Why are we raising an exception if the process is running? I'm guessing you meant 
```suggestion
        if not self.is_running():
            raise MongoDbProcessException
```

This check is very likely to have a race condition. The following is possible:

1. The mongo process is started.
1. Monkey Island checks if the process is running. It is, so it runs as planned.
1. The mongo process runs into an issue and crashes
1. Monkey Island freezes while it waits for mongodb, because it doesn't know the process crashed.

Since this doesn't really address the root issue, we'll need to come up with another solution. Doing this check in `_wait_for_mongodb_server()` is a better approach. We can pass the MongoDbProcess object to `connect_to_mongodb()` We can pass `none` if Monkey Island didn't start Mongo and skip the check.